source/transactions/tests/README.rst

```diff
@@ -151,6 +151,12 @@ Each YAML file has the following keys:
       multi-document read. If the operation is expected to return an error, the
       ``result`` is a single document that has one or more of the following
       fields:
+      
+      - ``clientError``: A boolean indicating whether the operation is
+        expected to fail due to a client side error (such as a document
+        being too large, or containing invalid keys). If the ``clientError``
+        key is not present but any of the other error keys are present,
+        the error is expected to have come from the server.
```

Comment:
Happy to change this to whatever the consensus ends up being.

Reply:
Thinking about this some more, I wonder if a `clientError` assertion is even necessary. I would think that the combination of a malformed insert document _and_ the session state _not_ transitioning would be enough to guarantee compliance with [SPEC-1195](https://jira.mongodb.org/browse/SPEC-1195). We could even couple this with another test that asserts that the session state _is_ advanced after encountering a non-client error (e.g. socket error using a fail point).

This invites an extra spec test, but would save all drivers from having to implement more test runner logic. If so, I might propose renaming `errors-client.*` to `errors-session_state.*` or something to that effect that indicates all tests are related to session state transitions (or lack thereof).

Reply:
Bumping this, as I'm still dubious that we need new spec test syntax to differentiate a client error from any other error. The non-negotiable dependency of these tests seems to be some facility to query the session for its transaction state and assert that it _hasn't_ changed.

To build upon this, I suppose the new test file could also include a test that asserts the state change for a non-client error. Again, we wouldn't need to inspect the error itself -- simply configure the fail point to trigger a runtime error and rely on the state assertion. 

Does anyone else see the value in differentiating the error type? My objective here is to limiting the amount of new syntax we introduce unless it's absolutely necessary.

Reply:
I agree with Jeremy here although I think it would be more readable to add his proposed test in `errors-client.yml` alongside the other tests rather than in a new file.